### PR TITLE
Fix python typing errors in tasks directory

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Fix a few typing issues in the `tasks` directory.

### Motivation

Every time I open a task file my IDE complains about hundreds of typing errors.
Those typing error can often reveal logic errors.

### Additional Notes

Main kinds of errors were:
- typing hints being just incorrect, eg: `build_tags: list[str] = None`
- using `dict.get`, `ctx.run`, or `regex.search` methods, which can return `None`, but assume the result not to be `None`
- wrong parameter / return type

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

File: tasks/go.py [hunk 1]
```diff
@@ -504,6 +505,7 @@ def add_go_module(path):
     modulespy_regex = re.compile(r"DEFAULT_MODULES = {\n(.+?)\n}", re.DOTALL | re.MULTILINE)
 
     all_modules_match = modulespy_regex.search(modulespy)
+    assert all_modules_match, "Could not find DEFAdULT_MODULES in modules.py"
```
Comment (GustavoCaso) [inline tasks/go.py:508]:
```suggestion
    assert all_modules_match, "Could not find DEFAULT_MODULES in modules.py"
```

?

**Review by GustavoCaso — APPROVED** [review]
LGTM.

File: tasks/go.py (hunk 1 above)
Comment (pgimalac) [reply tasks/go.py:508]: Yep good catch clearly a misstype 🙃

**Review by spencergilbert — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]
Thanks! I'll add mypy one day to the linters

**Pull request merged**